repo.go

```diff
@@ -467,11 +467,22 @@ func (r *Repo) RevokeKeyWithExpires(keyRole, id string, expires time.Time) error
 		return ErrKeyNotFound{keyRole, id}
```

Comment:
I understand this might be a little bit unrelated to this PR. 

But, the logic of the previous loop as well as line 466 is unreadable -- basically we are creating a new slice which excludes all the KeyID s related to the key we are removing. Why not deleting them from the list? -- and I do not think performance really matters here. 

I am open to other ideas but the current state unreadable.

Reply:
oh that's a good idea. it definitely took me a very careful read to understand the flow.